src/Microsoft.Health.Dicom.Core/Exceptions/InvalidQueryStringValuesException.cs

```diff
@@ -1,16 +1,17 @@
-﻿// -------------------------------------------------------------------------------------------------
+// -------------------------------------------------------------------------------------------------
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
 
 using System.Globalization;
+using System.Web;
 
 namespace Microsoft.Health.Dicom.Core.Exceptions;
 
 public class InvalidQueryStringValuesException : ValidationException
 {
     public InvalidQueryStringValuesException(string key, string error)
-        : base(string.Format(CultureInfo.InvariantCulture, DicomCoreResource.InvalidQueryStringValue, key, error))
+        : base(string.Format(CultureInfo.InvariantCulture, DicomCoreResource.InvalidQueryStringValue, key, HttpUtility.UrlEncode(error)))
```

Comment:
Shouldn't key be encoded then?

Reply:
no, the key is the part of the url path so it wouldn't match our router and would return a 404 (ex. in https://localhost:63838/v2/studies/1.2.332323232 `studies` is the key and if it were changed to something else we would just return a 404 instead of saying the key is invalid.)